### PR TITLE
fix: include auth sogs headers for join room and file download

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/session/apis/open_group_api/sogsv3/sogsV3FetchFile.ts [hunk 1]
```diff
@@ -168,7 +168,7 @@ const sogsV3FetchPreview = async (
     headers: null,
     serverPubkey: roomInfos.serverPublicKey,
     serverUrl: roomInfos.serverUrl,
-    doNotIncludeOurSogsHeaders: true,
+    doNotIncludeOurSogsHeaders: true, // the preview fetch is done before we joined a room, and so needs to not have any authentification
```
Comment (yougotwill) [inline ts/session/apis/open_group_api/sogsv3/sogsV3FetchFile.ts:171]: * and so doesn't need authentication

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/session/apis/open_group_api/sogsv3/sogsV3FetchFile.ts (hunk 1 above)
Comment (Bilb) [reply ts/session/apis/open_group_api/sogsv3/sogsV3FetchFile.ts]: fixed

**Review by yougotwill — APPROVED** [review]
LGTM

File: ts/session/apis/open_group_api/opengroupV2/ApiUtil.ts [hunk 2]
```diff
@@ -142,36 +142,33 @@ const defaultServerPublicKey = 'a03c383cf63c3c4efe67acc52112a6dd734b3a946b9545f4
...
-              completeUrl: getCompleteUrlFromRoom({
-                serverUrl: roomInfos.serverUrl,
-                serverPublicKey: roomInfos.serverPublicKey,
-                roomId: room.id,
-              }),
-            };
-          });
-        } catch (e) {
-          window?.log?.warn('loadDefaultRoomloadDefaultRoomssIfNeeded failed', e);
+  allowOnlyOneAtATime('loadDefaultRoomsSingle', async () => {
```
Comment (yougotwill) [inline ts/session/apis/open_group_api/opengroupV2/ApiUtil.ts:145]: We have lost the Promise return typing on the async function?

File: ts/session/apis/seed_node_api/SeedNodeAPI.ts [hunk 3]
```diff
@@ -146,9 +146,7 @@ export interface SnodeFromSeed {
 }
 
 const getSnodeListFromSeednodeOneAtAtime = async (seedNodes: Array<string>) =>
-  allowOnlyOneAtATime('getSnodeListFromSeednode', () =>
-    getSnodeListFromSeednode(seedNodes)
-  ) as Promise<Array<SnodeFromSeed>>;
+  allowOnlyOneAtATime('getSnodeListFromSeednode', () => getSnodeListFromSeednode(seedNodes));
```
Comment (yougotwill) [inline ts/session/apis/seed_node_api/SeedNodeAPI.ts:149]: Same thing here

File: ts/session/apis/open_group_api/opengroupV2/ApiUtil.ts (hunk 2 above)
Comment (Bilb) [reply ts/session/apis/open_group_api/opengroupV2/ApiUtil.ts:145]: it's a generic now and correctly typed ! thanks to https://type-level-typescript.com/

File: ts/session/apis/seed_node_api/SeedNodeAPI.ts (hunk 3 above)
Comment (Bilb) [reply ts/session/apis/seed_node_api/SeedNodeAPI.ts:149]: it's a generic now and correctly typed ! thanks to https://type-level-typescript.com/

File: ts/updater/updater.ts [hunk 4]
```diff
@@ -46,7 +46,13 @@ export async function start(
   }, 1000 * 60 * 10); // trigger and try to update every 10 minutes to let the file gets downloaded if we are updating
   stopped = false;
 
-  await checkForUpdates(getMainWindow, messages, logger);
+  global.setTimeout(async () => {
```
Comment (yougotwill) [inline ts/updater/updater.ts:49]: Doesn't this only need to run once? Can you explain why we have two intervals now?

Comment (Bilb) [reply ts/updater/updater.ts:49]: we tick every 10 minutes, and the first tick 2 minutes after start. there is only one interval

Comment (Bilb) [reply ts/updater/updater.ts:49]: we tick every 10 minutes, and the first tick 2 minutes after start. there is only one interval

Comment (yougotwill) [reply ts/updater/updater.ts:49]: Cool. LGTM

**Review by yougotwill — APPROVED** [review]
LGTM

**Review by yougotwill — APPROVED** [review]
LGTM

**Pull request merged**